src/middlewared/middlewared/plugins/cache.py

```diff
@@ -26,18 +27,36 @@ def get(self, key):
         Raises:
             KeyError: not found in the cache
         """
+        if isinstance(self.__cache[key][-1], datetime):
+            self.get_timeout(key)
+            return self.__cache[key][0]
+
         return self.__cache[key]
 
-    @accepts(Str('key'), Any('value'))
-    def put(self, key, value):
+    @accepts(Str('key'), Any('value'), Datetime('timeout', required=False))
+    def put(self, key, value, timeout):
```

Comment:
I think you need timeout=None, to not break existing API.

Reply:
Doesn't required=False set timeout=None in the validate method of the Datetime schema?

Reply:
My bad, I thought it would complain about missing parameter.